tests/DecodeChunkedStreamTest.php

```diff
@@ -43,6 +43,16 @@ public function provideChunkedEncoding()
                 ["6\r\nWi\r\n", "ki\r\n5\r\npedia\r\ne\r\n in\r\n\r\nchunks.\r\n0\r\n\r\n"],
                 "Wi\r\nkipedia in\r\n\r\nchunks."
             ],
+            'varnish-type-response-1' => [
+                ["0017\r\nWikipedia in\r\n\r\nchunks.\r\n0\r\n\r\n"]
+            ],
+            'varnish-type-response-2' => [
+                ["004\r\nWiki\r\n005\r\npedia\r\n00e\r\n in\r\n\r\nchunks.\r\n0\r\n\r\n"]
+            ],
+            'varnish-type-response-extra-line' => [
+                ["006\r\nWi\r\nki\r\n005\r\npedia\r\n00e\r\n in\r\n\r\nchunks.\r\n0\r\n\r\n"],
+                "Wi\r\nkipedia in\r\n\r\nchunks."
+            ]
```

Comment:
While the `ltrim` will definitely handle multiple leading `0`'s, could you add examples with more and less leading `0`'s then two. Just to ensure we don't accidentally refactor it specifically to two in the future. (You could even add a test adding a random number of `0`'s.)